https://marc.info/?l=freeipmi-devel&m=166752785904895&w=2

From 3ec9097fc558c888be9268344ea94a4caf5f7883 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Fri, 4 Nov 2022 01:58:38 +0000
Subject: [PATCH 1/3] configure.ac: Fix -Wimplicit-function-declaration for
 printf

Clang 16 makes these errors fatal by default:
```
error: call to undeclared library function 'printf' with type 'int (const char *, ...)'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
```

printf(3) is provided by <stdio.h>, not <stdlib.h>.

Signed-off-by: Sam James <sam@gentoo.org>
--- a/configure.ac
+++ b/configure.ac
@@ -692,7 +692,7 @@ fi
 
 AM_CONDITIONAL(WITH_FREEIPMI_ARGP, test "$have_argp" = "no")
 
-AC_TRY_COMPILE([#include <stdlib.h>],
+AC_TRY_COMPILE([#include <stdio.h>],
                [printf("%s\n", __FUNCTION__);],
                [AC_DEFINE([HAVE_FUNCTION_MACRO], [1], [Define is you have __FUNCTION__])])
 
